### DISCUSSION TITLE
Retrieving credentials not working

### DISCUSSION THREAD

**Discussion by Whitytom** [flobz/psa_car_controller#242] [Q&A] [answered]
I found a fine walkthrough for installing on win10 and followed it (sorry, it's german): 

https://openwb.de/forum/viewtopic.php?f=5&t=1206&start=260

But after installation ran through fine I got following db problem:

> C:\Opel>python3 server.py -f test.json -c charge_config1.json --web-conf
> 
> C:\Users\Thomas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\dash_bootstrap_components\_table.py:5: UserWarning:
> The dash_html_components package is deprecated. Please replace
> "import dash_html_components as html" with "from dash import html"
>   import dash_html_components as html
> 2021-10-07 17:01:01,391 :: WARNING :: Can't get language
> C:\Opel\web\view\views.py:8: UserWarning:
> The dash_core_components package is deprecated. Please replace
> "import dash_core_components as dcc" with "from dash import dcc"
>   import dash_core_components as dcc
> C:\Opel\web\figures.py:5: UserWarning:
> The dash_table package is deprecated. Please replace
> "import dash_table" with "from dash import dash_table"
> 
> Also, if you're using any of the table format helpers (e.g. Group), replace
> "from dash_table.Format import Group" with
> "from dash.dash_table.Format import Group"
>   import dash_table
> 2021-10-07 17:01:06,912 :: INFO :: update_data
> Exception in thread Thread-1:
> Traceback (most recent call last):
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 973, in _bootstrap_inner
>     self.run()
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 910, in run
>     self._target(*self._args, **self._kwargs)
>   File "C:\Opel\web\app.py", line 45, in start_app
>     run(config_flask(*args, **kwargs))
>   File "C:\Opel\web\app.py", line 75, in config_flask
>     importlib.import_module(view)
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\importlib\__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "C:\Opel\web\view\views.py", line 381, in <module>
>     update_trips()
>   File "C:\Opel\web\view\views.py", line 237, in update_trips
>     conn = Database.get_db(update_callback=False)
>   File "C:\Opel\web\db.py", line 133, in get_db
>     Database.init_db(conn)
>   File "C:\Opel\web\db.py", line 113, in init_db
>     Database.backup(conn)
>   File "C:\Opel\web\db.py", line 82, in backup
>     back_conn = sqlite3.connect(f"info_backup_{datetime.now()}.db")
> sqlite3.OperationalError: unable to open database file
> 
> C:\Opel>python3 server.py -f test.json -c charge_config1.json --web-conf
> C:\Users\Thomas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\dash_bootstrap_components\_table.py:5: UserWarning:
> The dash_html_components package is deprecated. Please replace
> "import dash_html_components as html" with "from dash import html"
>   import dash_html_components as html
> 2021-10-07 17:02:34,409 :: WARNING :: Can't get language
> C:\Opel\web\view\views.py:8: UserWarning:
> The dash_core_components package is deprecated. Please replace
> "import dash_core_components as dcc" with "from dash import dcc"
>   import dash_core_components as dcc
> C:\Opel\web\figures.py:5: UserWarning:
> The dash_table package is deprecated. Please replace
> "import dash_table" with "from dash import dash_table"

with every following run it changed:

> Also, if you're using any of the table format helpers (e.g. Group), replace
> "from dash_table.Format import Group" with
> "from dash.dash_table.Format import Group"
>   import dash_table
> 2021-10-07 17:02:35,444 :: INFO :: update_data
> Exception in thread Thread-1:
> Traceback (most recent call last):
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 973, in _bootstrap_inner
>     self.run()
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 910, in run
>     self._target(*self._args, **self._kwargs)
>   File "C:\Opel\web\app.py", line 45, in start_app
>     run(config_flask(*args, **kwargs))
>   File "C:\Opel\web\app.py", line 75, in config_flask
>     importlib.import_module(view)
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\importlib\__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "C:\Opel\web\view\views.py", line 381, in <module>
>     update_trips()
>   File "C:\Opel\web\view\views.py", line 243, in update_trips
>     car = CONFIG.myp.vehicles_list[0]  # todo handle multiple car
> AttributeError: 'Config' object has no attribute 'myp'

What am I doing wrong?

**Comment by flobz** [comment]
sqlite3.OperationalError: unable to open database file

Is the error you need to fix. Please look on internet, it might a permission issue.

**Reply by Whitytom** [reply]
OK, I found a problem: the info_backup file was not written because the filename contained ":" in db.py line 82, So I replaced it with

 ```
           Filename = f"info_backup_{datetime.now()}.db"
           back_conn = sqlite3.connect(Filename.replace(":","_"))
```
Now the file is written one time at the beginning with the correct name, for example "info_backup_2021-10-11 19_05_23.456600.db", but I still get

> C:\Users\Thomas\Documents\Daten\Opel>python3 server.py -f test.json -c charge_config1.json --web-conf
> C:\Users\Thomas\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\dash_bootstrap_components_table.py:5: UserWarning:
> The dash_html_components package is deprecated. Please replace
> import dash_html_components as html with from dash import html
> import dash_html_components as html
> 2021-10-11 19:05:22,410 :: WARNING :: Can't get language
> C:\Users\Thomas\Documents\Daten\Opel\web\view\views.py:8: UserWarning:
> The dash_core_components package is deprecated. Please replace
> import dash_core_components as dcc with from dash import dcc
> import dash_core_components as dcc
> C:\Users\Thomas\Documents\Daten\Opel\web\figures.py:5: UserWarning:
> The dash_table package is deprecated. Please replace
> import dash_table with from dash import dash_table
> 
> Also, if you're using any of the table format helpers (e.g. Group), replace
> from dash_table.Format import Group with
> from dash.dash_table.Format import Group
> import dash_table
> 2021-10-11 19:05:23,362 :: INFO :: update_data
> 
> Exception in thread Thread-1:
> Traceback (most recent call last):
> File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 973, in _bootstrap_inner
> self.run()
> File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\threading.py", line 910, in run
> self._target(*self._args, **self.kwargs)
> File "C:\Users\Thomas\Documents\Daten\Opel\web\app.py", line 45, in start_app
> run(config_flask(*args, **kwargs))
> File "C:\Users\Thomas\Documents\Daten\Opel\web\app.py", line 75, in config_flask
> importlib.import_module(view)
> File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2032.0_x64__qbz5n2kfra8p0\lib\importlib_init.py", line 127, in import_module
> return _bootstrap._gcd_import(name[level:], package, level)
> File "", line 1030, in _gcd_import
> File "", line 1007, in _find_and_load
> File "", line 986, in _find_and_load_unlocked
> File "", line 680, in _load_unlocked
> File "", line 850, in exec_module
> File "", line 228, in _call_with_frames_removed
> File "C:\Users\Thomas\Documents\Daten\Opel\web\view\views.py", line 381, in
> update_trips()
> File "C:\Users\Thomas\Documents\Daten\Opel\web\view\views.py", line 243, in update_trips
> car = CONFIG.myp.vehicles_list[0] # todo handle multiple car
> AttributeError: 'Config' object has no attribute 'myp'

**Comment by Whitytom** [comment, marked as answer]
A downgrade to 2.5.1 with the above mentioned change helped!

**Reply by flobz** [reply]
you can try v2.5.3 now :)
